Modal.js

```diff
@@ -0,0 +1,155 @@
+export class Modal {
+
+    #id = 'modal-version-04/13/2024'
+
+    modal = `
+        <div class="modal is-active" id="${this.#id}">
+            <div class="modal-background"></div>
+            <button class="modal-close is-large" aria-label="close"></button>
+            <div class="modal-card">
+                <header class="modal-card-head">
+                    <p class="modal-card-title"></p>
+                </header>
+                <section class="modal-card-body">
+                    <div class="field" id="modal-card-field">
+                        <label class="label"></label>
+                    </div>
+                </section>
+                <footer class="modal-card-foot is-justify-content-end">
+                    <div class="buttons">
+                        <button class="button is-danger" id="modal-primary-button"></button>
+                        <button class="button is-info" id="modal-secondary-button"></button>
+                    </div>
+                </footer>
+            </div>
+        </div> 
+    `
+
+    input = `<input id="modal-input" class="input">`
+
+    constructor(message, title, primaryBtnText, secondaryBtnText) {
+
+        this.title = title ?? 'Attention' 
+        this.message = message ?? 'This is the default alert message. Have a great day!'
+        this.requireUserInput = false
+        this.primaryBtnText = primaryBtnText ?? 'OK'
+        this.secondaryBtnText = secondaryBtnText ?? 'Cancel'
+    }
+
+    #createTemplate() {
+        
+        const existingModal = document.getElementById(`${this.#id}`)
+        existingModal && this.#close()
+
+        const parser = new DOMParser()
+        const parsedContent = parser.parseFromString(this.modal, 'text/html')
+                
+        const modalTitle = parsedContent.querySelector('.modal-card-title')
+        modalTitle.innerText = this.title
+
+        const modalMessage = parsedContent.querySelector('.label')
+        modalMessage.innerText = this.message
+
+        if (this.requireUserInput) {
+
+            const parsedInput = parser.parseFromString(this.input, 'text/html')
+            const input = parsedInput.querySelector('input')
+            const parsedSection = parsedContent.getElementById('modal-card-field')
+            parsedSection.append(input)
+
+        }
+
+        const modalPrimaryButton = parsedContent.getElementById('modal-primary-button')
+        modalPrimaryButton.innerText = this.primaryBtnText
+
+        const modalSecondaryButton = parsedContent.getElementById('modal-secondary-button')
+        modalSecondaryButton.innerText = this.secondaryBtnText
+
+        const modalDiv = parsedContent.querySelector('.modal')
+        document.body.appendChild(modalDiv)
+
+        this.#createCloseEvents() // Listens for events to close the modal
+
+    }
+
+    prompt(funcToRunOnOkay, funcToRunOnCancel) {
```

Comment:
Renaming these parameters increases readability _so much!_

Reply:
You were right! I have to overcome my fear of choosing names that are more than ten characters long. Descriptions make the code way more serviceable.

Reply:
So true!